rust/lance-file/src/v2/writer.rs

```diff
@@ -358,6 +375,112 @@ impl FileWriter {
     }
 }
 
+/// Utility trait for converting EncodedBatch to Bytes using the
+/// lance file format
+pub trait EncodedBatchWriteExt {
+    /// Serializes into a lance file, including the schema
+    fn try_to_self_described_lance(&self) -> Result<Bytes>;
+    /// Serializes into a lance file, without the schema.
+    ///
+    /// The schema must be provided to deserialize the buffer
+    fn try_to_mini_lance(&self) -> Result<Bytes>;
```

Comment:
Out of curiosity, why `try_to_self_described_lance`/`try_to_mini_lance` vs something more descriptive like `try_to_self_with_schema`/`try_to_lance_no_schema`? Are mini/self-described common in data format parlance, or is there  eventually things other than schema we'll emit from mini vs. self-described?

Reply:
> Out of curiosity, why try_to_self_described_lance/try_to_mini_lance vs something more descriptive like

I could be argued either way.  My original goal was to describe the "why" and not the "what".  As a potential API user if I see "to_lance_with_schema" and "to_lance_no_schema" I might not know why I would pick one or the other.  Still, there are definitely advantages to being more explicit with what is happening.

"self described" -> the resulting buffer can be deserialized without any extra information.
"mini" -> the smallest possible serialization.

> Are mini/self-described common in data format parlance

Not that I'm aware of.

>  there eventually things other than schema we'll emit from mini vs. self-described?

I don't know of anything at the moment.